### PR TITLE
[Axom] New version numbers

### PR DESCRIPTION
**Pull request by white238**
Just adding the new releases, and very slight changes

### CODE REVIEW

File: var/spack/repos/builtin/packages/axom/package.py [hunk 1]
```diff
@@ -100,15 +102,16 @@ class Axom(CachedCMakePackage, CudaPackage):
...
     depends_on("dtcmp", when="+scr")
 
     depends_on("raja~openmp", when="+raja~openmp")
     depends_on("raja+openmp", when="+raja+openmp")
     depends_on("raja+cuda", when="+raja+cuda")
 
     with when('+umpire'):
-        depends_on('umpire@5.0.1:5')
+        depends_on('umpire@6.0.0:', when='@0.6.0:')
+        depends_on('umpire@5:5.0.1', when='@:0.5.0')
```
Comment (white238) [inline var/spack/repos/builtin/packages/axom/package.py:112]: Looking over previous change and the issue that it was referencing, Axom pre 0.6.0 worked with Umpire 5.0.0 and 5.0.1.  I thinks this change is more accurate.  

It also adds the newer umpire for axom@0.6.0+

File: var/spack/repos/builtin/packages/axom/package.py [hunk 2]
```diff
@@ -100,15 +102,16 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("lua", when="+lua")
 
     depends_on("scr", when="+scr")
-    depends_on("kvtree@master", when="+scr")
+    depends_on("kvtree@main", when="+scr")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/axom/package.py:107]: Great catch!

**Review by alalazo — CHANGES_REQUESTED** [review]
It seems Gitlab pipelines failed compiling the new version of `axom` on some template argument deduction error

**Comment by white238** [comment]
> It seems Gitlab pipelines failed compiling the new version of `axom` on some template argument deduction error

I found one error where it was defaulting to raja 0.13.0 which definitely doesn't work with the Axom versions.  This should fix that. Hopefully it's the only problem.

**Comment by white238** [comment]
> > It seems Gitlab pipelines failed compiling the new version of `axom` on some template argument deduction error
> 
> I found one error where it was defaulting to raja 0.13.0 which definitely doesn't work with the Axom versions. This should fix that. Hopefully it's the only problem.

... insert narrator's voice ... It was not.

**Comment by white238** [comment]
This now passes!

**Review by alalazo — APPROVED** [review]

**Pull request merged**